salt/cloud/clouds/ec2.py

```diff
@@ -4762,7 +4762,7 @@ def get_password_data(
             rsa_key = kwargs['key']
             pwdata = base64.b64decode(pwdata)
             if HAS_M2:
-                key = RSA.load_key_string(rsa_key)
+                key = RSA.load_key_string(rsa_key.encode())
```

Comment:
When you do not pass an explicit encoding to `.encode()`, Python 2 will use its default encoding (i.e. `ascii`). Are we certain that the key will only include ascii? If not, we should be using `salt.utils.stringutils.to_bytes(rsa_key)`.

Reply:
Yes, Keys are only ever ascii. We could actually make that an explicit ascii decode.